src/common/helper.ts

```diff
@@ -318,35 +320,56 @@ async function readProtocolStream(
 
   const fs = isNode ? await importFSModule() : null;
 
-  let eof = false;
   let fileHandle: import('fs').promises.FileHandle;
 
   if (path && fs) {
     fileHandle = await fs.promises.open(path, 'w');
   }
-  const bufs = [];
-  while (!eof) {
-    const response = await client.send('IO.read', { handle });
-    eof = response.eof;
-    const buf = Buffer.from(
-      response.data,
-      response.base64Encoded ? 'base64' : undefined
-    );
-    bufs.push(buf);
-    if (path && fs) {
-      await fs.promises.writeFile(fileHandle, buf);
+  const buffers = [];
+  for await (const chunk of readable) {
+    buffers.push(chunk);
+    if (fileHandle) {
+      await fs.promises.writeFile(fileHandle, chunk);
     }
   }
+
   if (path) await fileHandle.close();
-  await client.send('IO.close', { handle });
   let resultBuffer = null;
   try {
-    resultBuffer = Buffer.concat(bufs);
+    resultBuffer = Buffer.concat(buffers);
   } finally {
     return resultBuffer;
   }
 }
 
+async function getReadableFromProtocolStream(
+  client: CDPSession,
+  handle: string
+): Promise<Readable> {
+  if (!isNode) {
+    throw new Error('Cannot create a stream outside of Node.js environment.');
```

Comment:
In the future we can solve this with https://github.com/nodejs/node/pull/39062. Perhaps add a TODO with a link?